### ISSUE TITLE
mac address on result

### ISSUE THREAD

**Issue by ekuzu** [eviltik/evilscan#76]
Do you plan to add mac address of founded machines on results?

**Comment by eviltik** [comment]
Hi

Well. Not sure i can go under the TCP layer using nodejs without a third party module. Or perhaps i got the mac in node network layer, probably.

PR welcome ? :)

**Comment by eviltik** [comment]
Hi there, just thinking ... what is your use case ?

**Comment by michaelniemand** [comment]
We have this requirement as well. We need to distinguish if the physical machine changed. Would be a great addition!

**Comment by eviltik** [comment]
I perfectly understand the need. It make sense only on the same LAN segments. It not on the same LAN, if you try to get remote clients' MAC addresses, you'll end up with the address of a router more than likely.

Are we all agree about that ?

**Comment by michaelniemand** [comment]
oooooh damn, you're right. ICMP doesn't even contain the MAC. Phew, in that case it wouldn't make much sense at all, because it's not reliable

**Comment by eviltik** [comment]
Yep, but it can be usefull on the same LAN segment. @ekuzu what is your use case ? ty

**Comment by ekuzu** [comment]
Sorry for late response, We had same use case like @michaelniemand , MAC is the unique identifier for machine in private networks.  And of course it's just useful for local network

Example;
I want to detect all open 9100 ports in our network (Printers) and save them to a database (IP, MAC, CustomName) with their MAC, if a printer has change on his IP then i can detect it easily over MAC identifier in our database and set new IP for it.
